crates/adapters/src/lib.rs

```diff
@@ -111,6 +120,34 @@
 //!
 //! * a [`Catalog`] object, which stores dictionaries of input and output
 //!   streams of the circuit.
+//!
+//! # Fault tolerance
+//!
+//! This crate implements support for "fault tolerant" circuits, that is,
+//! circuits whose operation can resume gracefully after a crash.  A crash will
+//! not cause a fault tolerant circuit to drop input or process it more than
+//! once, or to drop output or produce duplicate output, or to corrupt its
+//! internal state.
+//!
+//! The form of fault tolerance implemented in this crate can recover from
+//! crashes that kill processes or disrupt networking, but not crashes that lose
+//! storage or corrupt computations.
+//!
+//! Fault tolerance requires:
+//!
+//! * Input to be divided into numbered steps that can be retrieved repeatedly
```

Comment:
The circuit step() function gets called once per transport step, which is counted using `transport::Step`. I added some comments to clarify this.

What common abstraction do you see? I don't think that the circuit step() function counts global steps.

Reply:
it kindof does, for the outer nodes (not in loops).
I am expecting that the step count for input nodes in a DBPS graph is always the same as your step number.

Reply:
Yes, I think that's true. If they count the steps, where do I get that count from?

Reply:
I don't think that the circuit necessarily keeps track of this.
@ryzhyk?
But even if it doesn't, it would be useful to clarify that these 3 notions are the same one.

Reply:
They are the same. I added a clarifying comment.